core/web/resolver/vrf.go

```diff
@@ -0,0 +1,139 @@
+package resolver
+
+import (
+	"github.com/graph-gophers/graphql-go"
+
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/vrfkey"
+)
+
+type VRFKeyResolver struct {
+	key vrfkey.KeyV2
+}
+
+func NewVRFKeyResolver(key vrfkey.KeyV2) VRFKeyResolver {
+	return VRFKeyResolver{key}
+}
+
+// ID returns the ID of the VRF key, which is the public key.
+func (k VRFKeyResolver) ID() graphql.ID {
+	return graphql.ID(k.key.PublicKey.String())
+}
+
+func (k VRFKeyResolver) Compressed() string {
+	return k.key.PublicKey.String()
+}
+
+func (k VRFKeyResolver) Uncompressed() string {
+	uncompressed, err := k.key.PublicKey.StringUncompressed()
+	if err != nil {
+		// TODO: figure out better approach
+		panic(err)
+	}
+	return uncompressed
+}
+
+func (k VRFKeyResolver) Hash() string {
+	// TODO: should we allow a panic here?
+	return k.key.PublicKey.MustHash().String()
```

Comment:
I don't know how would the GQL implementation react to this, but I don't think we should allow panics here.

Maybe the same concept as for `Uncompressed()` applies?

Reply:
I followed what I did for `Uncompressed()` and return an error string in the event that getting the hash fails for whatever reason. The logic behind this decision is the same. But unfortunately I see downsides to both approaches.